services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchFeaturesSpec.groovy

```diff
@@ -0,0 +1,118 @@
+        def updateSwFeaturesResponse = northbound.updateSwitchFeatures(sw.dpId, switchFeatures)
+
+        then: "Correct response is returned"
+        updateSwFeaturesResponse.supportBfd == newSupportBfd
+        updateSwFeaturesResponse.supportMeters == newSupportMeters
+        updateSwFeaturesResponse.supportedTransitEncapsulation.sort() == newTransitEncapsulation
+
+        and: "Switch feature is really updated"
+        with(northbound.getSwitchFeatures(sw.dpId)) {
+            it.supportBfd == newSupportBfd
+            it.supportMeters == newSupportBfd
+            it.supportedTransitEncapsulation.sort() == newTransitEncapsulation
+        }
+
+        cleanup: "Restore init switch features on the switch"
+        northbound.updateSwitchFeatures(sw.dpId, initSwitchFeatures)
+    }
+
+    def "Informative error is returned when trying to get/update switch features with non-existing id"() {
+        when: "Try to get switch features info for non-existing switch"
+        northbound.getSwitchFeatures(NON_EXISTENT_SWITCH_ID)
+
+        then: "Human readable error is returned"
+        def e = thrown(HttpClientErrorException)
+        e.statusCode == HttpStatus.NOT_FOUND
+        e.responseBodyAsString.to(MessageError).errorMessage ==
+                "Failed to found features for switch '$NON_EXISTENT_SWITCH_ID'."
+
+        when: "Try to update switch features info for non-existing switch"
+        def switchFeatures = new SwitchFeaturesDto()
+        switchFeatures.supportBfd = true
+        switchFeatures.supportMeters = true
+        switchFeatures.supportedTransitEncapsulation = [FlowEncapsulationType.VXLAN.toString()]
+        northbound.updateSwitchFeatures(NON_EXISTENT_SWITCH_ID, switchFeatures)
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.statusCode == HttpStatus.NOT_FOUND
+        exc.responseBodyAsString.to(MessageError).errorMessage ==
+                "Failed to update switch features for switch '$NON_EXISTENT_SWITCH_ID'"
+    }
+
+    @Ignore("is not working yet, 500 is returned")
+    def "Informative error is returned when trying to update switch features with incorrect information"() {
+        given: "A switch"
+        def sw = topology.activeSwitches.first()
+
+        when: "Try to update switch features info for non-existing switch"
+        def switchFeatures = new SwitchFeaturesDto()
+        switchFeatures.supportedTransitEncapsulation = ["test"]
+        northbound.updateSwitchFeatures(sw.dpId, switchFeatures)
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.statusCode == HttpStatus.BAD_REQUEST
+        exc.responseBodyAsString.to(MessageError).errorMessage == "Not supported"
+    }
+
+    def "Unable to create a transit_vlan flow in case switch feature doesn't support this type of encapsulation"() {
```

Comment:
Do we need more iterations here? Try empty array as `supportedTransitEncapsulation`, try having both etc.

Reply:
the `supportMeters/supportBfd` fields are ignored. 
I mean we can change value for these fields, but they are ignored by system

Reply:
I mean more iterations only for `supportedTransitEncapsulation`